src/Domain/Creators/BinaryCompactObjectHelpers.cpp

```diff
@@ -42,16 +42,18 @@ create_grid_anchors(const std::array<double, 3>& center_a,
 
 TimeDependentMapOptions::TimeDependentMapOptions(
     double initial_time, ExpansionMapOptions expansion_map_options,
-    std::array<double, 3> initial_angular_velocity,
-    std::array<double, 3> initial_size_values_A,
-    std::array<double, 3> initial_size_values_B, const size_t initial_l_max_A,
-    const size_t initial_l_max_B, const Options::Context& context)
+    RotationMapOptions rotation_options,
```

Comment:
I technically can, however, I had this little intermediate commit to just standardize the constructor so all options were taken the same to make the change to `std::optional`s easier. If it's ok with you, I'd like to just leave it to avoid a complicated rebase with conflicts because I make this change and then immediately change it again the next commit.

Reply:
Yea don't worry about the commit history here, an extra commit is fine too